### PR TITLE
nixos/syncthing: move configuration to condigDir

### LINKED ISSUE #47513: nixos/syncthing: bad default settings (perms 700 on home)
## Issue description
The NixOS syncthing service with default settings sets ```-home=$HOME``` (where $HOME is /var/lib/syncthing) and with that breaks upstream assumption: https://github.com/syncthing/syncthing/issues/3434#issuecomment-235401876.

The result is that you cannot run syncthing with (nixos) default settings and give access to various users with ACLs, as syncthing will set perms 700 on the directory passed in ```-home=``` each time it starts. And as ```-home=``` is the same as ```$HOME```, your default shared folders are now hidden behind 700 perms.

Workaround (hack):
```
  # Undo the 700 perms syncthing sets on /var/lib/syncthing on startup. Without
  # this the ACL entries stop working.
  # See https://github.com/syncthing/syncthing/issues/3434.
  systemd.services.syncthing.postStart = lib.mkForce ''
    sleep 10
    chmod g+rx "${config.services.syncthing.dataDir}"
  '';
```

Based on https://github.com/syncthing/syncthing/issues/3434#issuecomment-235401876, the fix seems to be simply setting ```-home=$HOME/.config/syncthing``` instead of ```-home=$HOME```.

### Steps to reproduce
1. Add ```services.syncthing.enable = true;``` and ```sudo nixos-rebuild switch```
2. Add ACL: ```sudo setfacl -m u:$USER:rx /var/lib/syncthing```
3. Check that your user has access: ```ls -l /var/lib/syncthing```
4. Restart syncthing: ```sudo systemctl restart syncthing```
5. You no longer have access to the share:
```
ls -l /var/lib/syncthing
ls: cannot open directory '/var/lib/syncthing/': Permission denied
```

## Technical details
 - system: `"x86_64-linux"`
 - host os: `Linux 4.14.70, NixOS, 18.03.133231.01f5e794913 (Impala)`
 - multi-user?: `yes`
 - sandbox: `yes`
 - version: `nix-env (Nix) 2.0.4`
 - channels(root): `"nixos-18.03.133231.01f5e794913"`
 - channels(bfo): `""`
 - nixpkgs: `/nix/var/nix/profiles/per-user/root/channels/nixos/nixpkgs`

### PR DESCRIPTION
**Pull request by rnhmjoj**
###### Motivation for this change
Fix #47513 

###### Things done
- [x] Tested using a NixOS VM
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

---

### CODE REVIEW

**Comment by bjornfor** [comment]
Would you mind adding a ref to "https://github.com/syncthing/syncthing/issues/3434#issuecomment-235401876" (or the issue itself) in the commit message?

**Review by bjornfor — APPROVED** [review]

**Comment by bjornfor** [comment]
Hm, is this too late to target 18.09? (Currently this PR updates 18.09 release notes.) I personally wouldn't mind seeing it land in 18.09, but it is short notice :-)

**Comment by rnhmjoj** [comment]
@bjornfor done. Thank you.

**Comment by rnhmjoj** [comment]
I moved the check to 19.03 for obvious reasons.

**Comment by rnhmjoj** [comment]
@joachifm done.

**Comment by rnhmjoj** [comment]
ping

**Comment by rnhmjoj** [comment]
Thank you all!

**Comment by bjornfor** [comment]
Thanks!

File: nixos/doc/manual/release-notes/rl-1903.xml [hunk 1]
```diff
@@ -97,6 +97,16 @@
        start org.nixos.nix-daemon</command>.
       </para>
      </listitem>
+     <listitem>
+      <para>
+        The Syncthing state and configuration data has been moved from
```
[the remark this answers is not in the document]
Comment (rnhmjoj) [reply nixos/doc/manual/release-notes/rl-1903.xml]: Ah, thank you. I'll make a PR to fix this soon.

**Pull request merged**